app/controllers/addresses_controller.rb

```diff
@@ -0,0 +1,51 @@
+class AddressesController < ApplicationController
+
+  def create
+    form_with_errors = false
+    addresses_params['address_forms'].each do |address_type, address_params|
+      address_form = instance_variable_set("@#{address_type}_address_form", UserAddressForm.from_params(address_params))
+      next unless params[address_type] == 'true'
+      if address_form.valid?
+        @address = current_user.addresses.find_by(id: address_params['id'])
+        @address = current_user.addresses.new unless @address.present?
+        @address.attributes = address_form.attributes.except('id')
+        @address.save
+      else
+        form_with_errors = true
+      end
```

Comment:
1. Can be separate method
2. next unless params[address_type] == 'true' should be done earlier
3. find_or_initialize_by or just create
4. This should process any address not only current user